lighthouse-core/audits/byte-efficiency/legacy-javascript.js

```diff
@@ -371,8 +441,21 @@ class LegacyJavascript extends Audit {
     const urlToMatchResults =
       this.detectAcrossScripts(matcher, artifacts.ScriptElements, networkRecords, bundles);
     urlToMatchResults.forEach((matches, url) => {
-      /** @type {typeof tableRows[number]} */
-      const row = {url, signals: [], locations: []};
+      // Only estimate savings if first party code has legacy code.
+      let wastedBytes = 0;
+      if (thirdPartyWeb.isFirstParty(url, mainDocumentEntity)) {
+        wastedBytes = this.estimateWastedBytes(matches);
```

Comment:
maybe `wastedBytes` should still be set for 3p items, and `wastedBytesByUrl` could set to zero for items that are 3p so it doesn't count towards estimated savings. would that be confusing in the table? def. need to get the 3p report filter to use third-party-web so that the default view only shows stuff that contributes to estimated 1p savings.

Oh, I should say: I'm thinking it's useful to still track legacy js from 3p and make it possible to see that in the report. I don't wanna lose that info. just don't want it to count, and 3p should be hidden by default

Reply:
gotcha yeah I think that all makes sense 👍